src/fasttext.cc

```diff
@@ -647,33 +668,197 @@ void FastText::loadVectors(std::string filename) {
 void FastText::train(const Args args) {
   args_ = std::make_shared<Args>(args);
   dict_ = std::make_shared<Dictionary>(args_);
+
   if (args_->input == "-") {
     // manage expectations
     throw std::invalid_argument("Cannot use stdin for training!");
   }
-  std::ifstream ifs(args_->input);
-  if (!ifs.is_open()) {
-    throw std::invalid_argument(
-        args_->input + " cannot be opened for training!");
-  }
-  dict_->readFromFile(ifs);
-  ifs.close();
 
-  if (args_->pretrainedVectors.size() != 0) {
-    loadVectors(args_->pretrainedVectors);
-  } else {
-    input_ = std::make_shared<Matrix>(dict_->nwords()+args_->bucket, args_->dim);
-    input_->uniform(1.0 / args_->dim);
-  }
+  if (!args.incr) {
+    std::ifstream ifs(args_->input);
```

Comment:
Is there a reason you don't use std::ifstream::binary similar to https://github.com/facebookresearch/fastText/blob/1e6637513bdca926e83e540b0a5296dd0878adca/src/fasttext.cc#L180? this causes the data to be read incorrectly without the binary flag.